db/migrations/20230713155553_endpoint_versions.sql

```diff
@@ -0,0 +1,17 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TABLE unweave.endpoint_version (
+    id text NOT NULL PRIMARY KEY,
+    parent_endpoint_id text NOT NULL REFERENCES unweave.endpoint (id),
+    endpoint_id text NOT NULL REFERENCES unweave.endpoint (id),
```

Comment:
What is the difference between a parent_endpoint and and endpoint? Is this what you meant in the PR description by "And endpoint version is an endpoint"? This is somewhat confusing. Could you not treat an endpoint as a "grouping" and a version as a concrete URL? For example, in Cloud Run, an App is a grouping of revisions. An App itself doesn't serve any traffic. A _revision_ serves traffic.

Reply:
I agree with @noorvir. The polymorphism seems confusing. `Endpoint` and `EndpointVersion` seem like meaningfully different entities in our domain: an EndpointVersion serves traffic, and an Endpoint routes traffic to a particular version.